packages/flame/lib/src/effects/provider_interfaces.dart

```diff
@@ -0,0 +1,7 @@
+import 'package:vector_math/vector_math_64.dart';
+
+/// Interface for a component that can be affected by move effects.
+abstract class PositionProvider {
```

Comment:
Will there be one interface for each part of the transform matrix? It will be very bloated with like `implements PositionProvider, ScaleProvoder, AngleProvider` everywhere where it is needed? Or will there be a mixin or that implements all of them that can be used?